src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -1296,6 +1296,9 @@ bool PhaseIdealLoop::loop_predication_impl_helper(IdealLoopTree* loop, IfProjNod
   invar.map_ctrl(if_success_proj, new_predicate_proj); // so that invariance test can be appropriate
 
   // Eliminate the old If in the loop body
+  // If a range check is eliminated, data dependent nodes (Load and range check CastII nodes) are now dependent on 2
+  // range check predicates (one for the start of the loop, one for the end) but we can only keep track of one control
```

Comment:
To follow the naming conventions added by the changes around JDK-8288981:
```suggestion
  // Hoisted Check Predicates (one for the start of the loop, one for the end) but we can only keep track of one control
```